File: device.go
package shipyard

// Device represents a physical AV device and its configuration
type Device struct {
	ID                 string                           `json:"id"`
	Room               string                           `json:"room"`
	APIControllable    bool                             `json:"api_controllable"`
	PublicDescription  string                           `json:"publicDescription"`
	PrivateDescription string                           `json:"privateDescription"`
	Address            string                           `json:"address"`
	Commands           map[string]GenericControlRequest `json:"commands"`
	Presets            map[string]string                `json:"presets,omitempty"`
	Driver             string                           `json:"driver"`
	DyanmicPorts       bool                             `json:"dynamicPorts"`
	Ports              []DevicePort                     `json:"ports"`
	Tags               Tags                             `json:"tags"`
}

// DeviceTemplate represents a template for a device
type DeviceTemplate struct {
	ID       string `json:"id"`
	Template Device `json:"template"`
}

// DevicePort represents a port on a physical device
type DevicePort struct {
	ID        string         `json:"id"`
	Name      string         `json:"name"`
	Endpoints []PortEndpoint `json:"endpoints"`
	Incoming  bool           `json:"incoming"`
	Type      string         `json:"type"`
}

type PortEndpoint struct {
	Device string `json:"device"`
	Port   string `json:"port"`
}

// Tags are a set of arbitrary key value pairs used for storing extra information
type Tags map[string]string

File: ui-old/src/app/components/room-page/devices/device.ts
import { Port } from "../ports/port";

export class Device {
    id: string;
    // api_controllable: boolean;
    publicDescription: string;
    // privateDescription: string;
    address: string;
    driver: string;
    dynamicPorts: boolean;
    ports: Port[];
    tags: Map<string, string>;
  
    Type: string;

    constructor() {
        this.tags = new Map();
        this.id = "";
        this.publicDescription = "";
        this.address = "";
        this.driver = "";
        this.dynamicPorts = true;
        this.ports = [];
        this.Type = "";
    }
}